dev/com.ibm.ws.jpa.container/src/com/ibm/ws/jpa/container/osgi/internal/JPAComponentImpl.java

```diff
@@ -379,6 +391,10 @@ public void applicationStarting(ApplicationInfo appInfo) {
                 FFDCFilter.processException(e, this.getClass().getName(), "457");
             }
         } finally {
+            if (setContext) {
+                ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().endContext();
```

Comment:
There any chance this might throw an exception?

Reply:
@jgrassel I don't believe so. Looking at the implementation, I don't see that it throws an exception. Also, there is a similar implementation in the `CDIRuntimeImpl` that I am reusing